pyladies_cz.py

```diff
@@ -164,13 +165,31 @@ def date_range(dates, sep='–'):
 
 
 def read_yaml(filename, default=MISSING):
+    """Read the given YAML file
+
+    If the file is not found:
+    - if default is given, return it
+    - else, raise FileNotFoundError
+    """
+
+    # Reading YAML is slow, and we need info from cities.yml for every single
+    # page. To avoid loading all the time, we cache the result.
+    # To make the site react to changes in the files, we use the file size
+    # and last-modified time as part of the cache key. If either changes,
+    # the cache will be invalidated and the file will be read from disk again.
+
     try:
-        file = open(filename, encoding='utf-8')
+        info = os.stat(filename)
     except FileNotFoundError:
         if default is MISSING:
             raise
         return default
-    with file:
+    return _read_yaml_cached(filename, info.st_size, info.st_mtime)
```

Comment:
Juts curious. Why size AND mtime?

Reply:
That's what Python does for `pyc` files, so I know it works well :)

Neither is perfect, and since `os.stat` returns both, using both is not much slower than just one of them.
You could also ask *why not more?* -- the other info from the [os.stat result](https://docs.python.org/3/library/os.html?highlight=stat#os.stat_result) either doesn't work well on all platforms, or is essentially redundant.

Reply:
Thanks.